### PR TITLE
Add support for Slack follow up

### PR DESCRIPTION
**Pull request by lasryaric**
Adding support for Slack follow up conversation:
Talking to the bot multiple times in the same thread would end up posting user message in the same conversation.

ContentFragment will be in a follow up PR, which will be highly facilitated by this one.

### CODE REVIEW

File: connectors/src/connectors/slack/bot.ts [hunk 1]
```diff
@@ -245,37 +262,55 @@ async function botAnswerMessage(
...
-        timezone: slackChatBotMessage.slackTimezone || "Europe/Paris",
-        username: slackChatBotMessage.slackUserName,
-        fullName:
-          slackChatBotMessage.slackFullName ||
-          slackChatBotMessage.slackUserName,
-        email: slackChatBotMessage.slackEmail,
-        profilePictureUrl: slackChatBotMessage.slackAvatar || null,
-      },
+
+  const messageReqBody = {
```
Comment (spolu) [inline connectors/src/connectors/slack/bot.ts:266]: we pass all mentions now?

File: connectors/src/connectors/slack/bot.ts [hunk 2]
```diff
@@ -245,37 +262,55 @@ async function botAnswerMessage(
...
+    slackChatBotMessage.conversationId = conversationId;
+    await slackChatBotMessage.save();
   }
 
-  const conversation = convRes.value;
-
-  slackChatBotMessage.conversationId = conversation.sId;
-  await slackChatBotMessage.save();
-
+  const conversationRes = await dustAPI.getConversation({ conversationId });
```
Comment (spolu) [inline connectors/src/connectors/slack/bot.ts:309]: You don't need to getConversation if you createConversation. Only do so if you postUserMessage

File: connectors/src/lib/dust_api.ts [hunk 3]
```diff
@@ -381,22 +381,49 @@ export class DustAPI {
...
+    const json = await res.json();
+    if (json.error) {
+      return new Err(json.error as DustAPIErrorResponse);
+    }
+
+    return new Ok(json.message as UserMessageType);
+  }
+
+  async streamAgentMessageEvents({
+    conversationId,
```
Comment (spolu) [inline connectors/src/lib/dust_api.ts:414]: why change the signature here?

File: connectors/src/connectors/slack/bot.ts (hunk 1 above)
Comment (lasryaric) [reply connectors/src/connectors/slack/bot.ts:266]: We return an error if there is more than one mention.

File: connectors/src/connectors/slack/bot.ts (hunk 2 above)
Comment (lasryaric) [reply connectors/src/connectors/slack/bot.ts:309]: done

File: connectors/src/lib/dust_api.ts (hunk 3 above)
Comment (lasryaric) [reply connectors/src/lib/dust_api.ts:414]: I initially thought I was going to have a code path where I had the message id but not the agentMessageType itself.
That being said, it's an API client and the API only needs the id, so having the ConversationType and AgentMessageType make using this API a bit harder.

Comment (spolu) [reply connectors/src/lib/dust_api.ts:414]: More restrictive but cleaner let's not change it if not needed.

File: connectors/src/connectors/slack/bot.ts (hunk 1 above)
Comment (spolu) [reply connectors/src/connectors/slack/bot.ts:267]: We should not we said we wanted to treat them as options not erroring?

Comment (spolu) [reply connectors/src/connectors/slack/bot.ts:267]: tbh I don't see the new error handling but I see a slice(0, 1) gone?

**Review by spolu — COMMENTED** [review]
Replied to your comments

File: connectors/src/connectors/slack/bot.ts (hunk 1 above)
Comment (lasryaric) [reply connectors/src/connectors/slack/bot.ts:267]: We agreed on that [yesterday](https://dust4ai.slack.com/archives/C050SM8NSPK/p1696523341985719?thread_ts=1696517709.263099&cid=C050SM8NSPK).
The removed slice() was a left over from yesterday, it should not have been there.

The error handling for more than one mention is [here](https://github.com/dust-tt/dust/blob/main/connectors/src/connectors/slack/bot.ts#L198-L203).

**Review by spolu — DISMISSED** [review]

File: connectors/src/lib/dust_api.ts (hunk 3 above)
Comment (lasryaric) [reply connectors/src/lib/dust_api.ts:414]: Sounds good, I will fix it before merging. I am working on content fragment right now.

**Pull request merged**